pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+        """Return the ambient color of the light."""
+        return self.GetAmbientColor()
+
+    @ambient_color.setter
+    def ambient_color(self, color):
+        """Set the ambient color of the light."""
+        self.SetAmbientColor(color)
+
+    @property
+    def diffuse_color(self):
+        """Return the diffuse color of the light."""
+        return self.GetDiffuseColor()
+
+    @diffuse_color.setter
+    def diffuse_color(self, color):
+        """Set the diffuse color of the light."""
+        self.SetDiffuseColor(color)
+
+    @property
+    def specular_color(self):
+        """Return the specular color of the light."""
+        return self.GetSpecularColor()
+
+    @specular_color.setter
+    def specular_color(self, color):
+        """Set the specular color of the light."""
+        self.SetSpecularColor(color)
+
+    # TODO: implement light.color = ... using setattr? no guarantee that a getter would always make sense, so property won't work!
+
+    @property
+    def position(self):
+        """Return the position of the light.
+
+        Note: the position is defined in the coordinate space indicated
+        by the light's transformation matrix (if it exists). To get the
+        light's world space position, use the ``world_position`` property.
+
+        """
+        return self.GetPosition()
+
+    @position.setter
+    def position(self, pos):
+        self.SetPosition(pos)
+
+    @property
+    def world_position(self):
+        # TODO: is this name and configuration OK? Same for world_focal_point
+        # TODO: can a transformation matrix happen accidentally? If not, perhaps we can just not expose these at all!
+        """Return the world space position of the light."""
+        return self.GetTransformedPosition()
+
+    @property
+    def focal_point(self):
+        """Return the focal point of the light.
+
+        Note: the focal point is defined in the coordinate space indicated
+        by the light's transformation matrix (if it exists). To get the
+        light's world space focal point, use the ``world_focal_point``
+        property.
+
+        """
+        return self.GetFocalPoint()
+
+    @focal_point.setter
+    def focal_point(self, pos):
+        self.SetFocalPoint(pos)
+
+    @property
+    def world_focal_point(self):
+        """Return the world space focal point of the light."""
+        return self.GetTransformedFocalPoint()
+
+    @property
+    def intensity(self):
+        """Return the brightness of the light (between 0 and 1)."""
+        return self.GetIntensity()
+
+    @intensity.setter
+    def intensity(self, intensity):
+        self.SetIntensity(intensity)
+
+    @property
+    def is_on(self):
+        # TODO: is this name OK? Just "on" sounds too short.
+        """Return whether the light is on."""
+        return bool(self.GetSwitch())
+
+    @is_on.setter
+    def is_on(self, state):
+        """Set whether the light should be on."""
+        self.SetSwitch(state)
```

Comment:
```suggestion
    @property
    def on(self):
        """Return ``True`` when the light is on."""
        return bool(self.GetSwitch())

    @is_on.setter
    def on(self, state):
        """Turn the light on or off"""
        self.SetSwitch(state)
```

`numpy` has `T` for transpose, so honestly I think we're fine.  The ``is`` is redundant and doesn't make sense in the setter.